src/test/unit/engine/InputTest.cpp

```diff
@@ -165,6 +169,548 @@ TEST(InputTest, Links)
+ */
+class L2TestRegion: public TestRegionBase
+{
+public:
+  L2TestRegion(const ValueMap& params, Region *region) :
+    TestRegionBase(params, region)
+  {
+  }
+
+  L2TestRegion(BundleIO& bundle, Region* region) :
+    TestRegionBase(bundle, region)
+  {
+  }
+
+  L2TestRegion(capnp::AnyPointer::Reader& proto, Region* region) :
+    TestRegionBase(proto, region)
+  {
+  }
+
+  virtual ~L2TestRegion()
+  {
+  }
+
+  std::string getNodeType()
+  {
+    return "L2TestRegion";
+  }
+
+  // Used by RegionImplFactory to create and cache
+  // a nodespec. Ownership is transferred to the caller.
+  static Spec* createSpec()
+  {
+    auto ns = new Spec;
+
+    /* ----- inputs ------- */
+    ns->inputs.add(
+      "feedForwardIn",
+      InputSpec(
+        "Feed-forward input for the node",
+        NTA_BasicType_UInt64,
+        0, // count. omit?
+        true, // required?
+        false, // isRegionLevel,
+        false  // isDefaultInput
+        ));
+
+    ns->inputs.add(
+      "lateralIn",
+      InputSpec(
+        "Lateral input for the node",
+        NTA_BasicType_UInt64,
+        0, // count. omit?
+        true, // required?
+        false, // isRegionLevel,
+        false  // isDefaultInput
+        ));
+
+    /* ----- outputs ------ */
+    ns->outputs.add(
+      "out",
+      OutputSpec(
+        "Primary output for the node",
+        NTA_BasicType_UInt64,
+        3, // 1st is output; 2nd is the given feedForwardIn; 3rd is lateralIn
+        false, // isRegionLevel
+        true // isDefaultOutput
+        ));
+
+    return ns;
+  }
+
+  /**
+   * Inputs/Outputs are made available in initialize()
+   * It is always called after the constructor (or load from serialized state)
+   */
+  void initialize() override
+  {
+    nodeCount_ = getDimensions().getCount();
+    out_ = getOutput("out");
+    feedForwardIn_ = getInput("feedForwardIn");
+    lateralIn_ = getInput("lateralIn");
+  }
+
+  // Compute outputs from inputs and internal state
+  void compute() override
+  {
+    NTA_DEBUG << "> Computing: " << getName() << " <";
+
+    const Array & outputArray = out_->getData();
+    NTA_CHECK(outputArray.getCount() == 3);
+    NTA_CHECK(outputArray.getType() == NTA_BasicType_UInt64);
+    UInt64 *baseOutputBuffer = (UInt64*)outputArray.getBuffer();
+
+    std::vector<UInt64> ffInput;
+    feedForwardIn_->getInputForNode(0, ffInput);
+    NTA_CHECK(ffInput.size() > 1);
+
+    NTA_DEBUG << getName() << ".compute: ffInput size=" << ffInput.size()
+              << "; inputValue=" << ffInput[0];
+
+    std::vector<UInt64> latInput;
+    lateralIn_->getInputForNode(0, latInput);
+    NTA_CHECK(latInput.size() > 1);
+
+    NTA_DEBUG << getName() << ".compute: latInput size=" << latInput.size()
+              << "; inputValue=" << latInput[0];
+
+    baseOutputBuffer[0] = ffInput[0] + latInput[0];
+    baseOutputBuffer[1] = ffInput[0];
+    baseOutputBuffer[2] = latInput[0];
```

Comment:
Adding some comments would help. How about something like
"Only the first element of baseOutputBuffer represents region output. We keep track of inputs to the region using the rest of the baseOutputBuffer vector. These inputs are used in the tests"